### ISSUE TITLE
How do I specify my own RGB value?

### ISSUE THREAD

**Issue by Joe23232** [nannou-org/nannou#677]
I am kinda a noob to Rust, I am having trouble figuring out how to specify my very specific RGB value for `nannou`. How would I do this?

**Comment by MacTuitui** [comment]
`let color = srgb(0.2,0.5,0.0)` will give you a color. You also have `srgb8(125,255,12)` if you prefer.

**Comment by Joe23232** [comment]
So I assume that `srgB` is the integer version?

**Comment by MacTuitui** [comment]
Yes! It's actually an eight, like in `u8`, the unsigned char type that you would use to specify a color from 0 to 255.

You also have the `srgba(red, green, blue, alpha)` variant if you want to specify the alpha value as well.

**Comment by Joe23232** [comment]
Ah I see, thanks mate :)

**Comment by MacTuitui** [comment]
But if you start to get serious about color and graphics, I can not recommend enough to get familiar with what `srgb` is, what are linear color spaces, what is the process to convert to RGB (and the relation with gamma correction).... and eventually you'll find out that nannou might be the only framework that does the right thing. You have floating point color precision in a linear space. So all your color operations are *right* out of the box, and you can reach far better results thanks to that.

**Comment by Joe23232** [comment]
> But if you start to get serious about color and graphics, I can not recommend enough to get familiar with what `srgb` is, what are linear color spaces, what is the process to convert to RGB (and the relation with gamma correction).... and eventually you'll find out that nannou might be the only framework that does the right thing. You have floating point color precision in a linear space. So all your color operations are _right_ out of the box, and you can reach far better results thanks to that.

So in other words it is simply more precise than the `srgB` version?

**Comment by MacTuitui** [comment]
Yes. Any color you use will internally be converted to the right format to use the available precision. You can sitll use `srgb8` to specify the color you want (and I still do that all the time): the details are when two colors mix together.

**Comment by Joe23232** [comment]
Ah I see, thanks mate

**Comment by mitchmindtree** [comment]
Thanks both! It looks like this is resolved, so closing this for now.
